tests/Avalonia.Controls.UnitTests/Automation/ControlAutomationPeerTests.cs

```diff
@@ -0,0 +1,253 @@
+﻿using System.Linq;
+using Avalonia.Automation.Peers;
+using Avalonia.Automation.Platform;
+using Avalonia.Automation.Provider;
+using Avalonia.Controls.Presenters;
+using Avalonia.Controls.Primitives;
+using Avalonia.Controls.Templates;
+using Avalonia.Platform;
+using Avalonia.UnitTests;
+using Avalonia.VisualTree;
+using Moq;
+using Xunit;
+
+namespace Avalonia.Controls.UnitTests.Automation
+{
+    public class ControlAutomationPeerTests
```

Comment:
Should we test that properties trickle down, e.g. AutomationProperties.Name ?

Reply:
I've not implemented `Name` yet, how should it "trickle down"?

Reply:
With "trickle down" I was thinking about trying to use control properties to get the name. E.g. Button should try to use it's content as string to see if it get's that name then. I think the following order is good:

1. Try `AutomationProperties.Name` to see if there is a name
2. Try to defer name from content (Button content or text of a TextBlock)
3. Try super.GetNameCore()
4. Return empty string as there was no way of getting actual name

Reply:
Actually, seems like I have implemented this: https://github.com/AvaloniaUI/Avalonia/blob/feature/ui-automation/src/Avalonia.Controls/Automation/Peers/ControlAutomationPeer.cs#L91

I think this should work ask you expect, where derived `AutomationPeer`s try to get the name from e.g. the button content, and if they fail they call this base implementation which checks the property.

Reply:
I see, yes that should work as expected. Seems fine to me.